builder/triton/builder.go

```diff
@@ -0,0 +1,110 @@
+package triton
+
+import (
+	"log"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/communicator"
+	"github.com/mitchellh/packer/helper/config"
+	"github.com/mitchellh/packer/packer"
+)
+
+const (
+	BuilderId = "joyent.triton"
+)
+
+type Builder struct {
+	config Config
+	runner multistep.Runner
+}
+
+func (b *Builder) Prepare(raws ...interface{}) ([]string, error) {
+	errs := &multierror.Error{}
+
+	err := config.Decode(&b.config, &config.DecodeOpts{
+		Interpolate:        true,
+		InterpolateContext: &b.config.ctx,
+	}, raws...)
+	if err != nil {
+		errs = multierror.Append(errs, err)
+	}
+
+	errs = multierror.Append(errs, b.config.AccessConfig.Prepare(&b.config.ctx)...)
+	errs = multierror.Append(errs, b.config.SourceMachineConfig.Prepare(&b.config.ctx)...)
+	errs = multierror.Append(errs, b.config.Comm.Prepare(&b.config.ctx)...)
+	errs = multierror.Append(errs, b.config.TargetImageConfig.Prepare(&b.config.ctx)...)
+
+	return nil, errs.ErrorOrNil()
+}
+
+func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packer.Artifact, error) {
+	config := b.config
+
+	driver, err := NewDriverTriton(ui, config)
+	if err != nil {
+		return nil, err
+	}
+
+	state := new(multistep.BasicStateBag)
+	state.Put("config", b.config)
+	state.Put("driver", driver)
+	state.Put("hook", hook)
+	state.Put("ui", ui)
+
+	steps := []multistep.Step{
+		&StepCreateSourceMachine{},
+		&communicator.StepConnect{
+			Config: &config.Comm,
+			Host:   commHost,
+			SSHConfig: sshConfig(
+				b.config.Comm.SSHAgentAuth,
+				b.config.Comm.SSHUsername,
+				b.config.Comm.SSHPrivateKey,
+				b.config.Comm.SSHPassword),
+		},
+		&common.StepProvision{},
+		&StepStopMachine{},
+		&StepCreateImageFromMachine{},
+		&StepDeleteMachine{},
+	}
+
+	if b.config.PackerDebug {
+		b.runner = &multistep.DebugRunner{
+			Steps:   steps,
+			PauseFn: common.MultistepDebugFn(ui),
+		}
+	} else {
+		b.runner = &multistep.BasicRunner{Steps: steps}
+	}
+
+	b.runner.Run(state)
```

Comment:
This section can be replaced with:

```golang
// Run
b.runner = common.NewRunnerWithPauseFn(steps, b.config.PackerConfig, ui, state)
b.runner.Run(state)
```

See [virtualbox/iso/builder.go#L283-L285](https://github.com/mitchellh/packer/blob/master/builder/virtualbox/iso/builder.go#L283-L285)